### PR TITLE
Make sure that the timeline comes back in the right order

### PR DESCRIPTION
**Pull request by richvdh**
It turns out that it does, and I was just being a crank. But no harm in testing
for it explicitly.

### CODE REVIEW

**Comment by richvdh** [comment]
The jenkins failure is a lie.

File: tests/31sync/04timeline.pl [hunk 1]
```diff
@@ -16,10 +16,16 @@
          ( $room_id ) = @_;
 
          matrix_send_room_text_message( $user, $room_id,
-            body => "A test message",
+            body => "Test message 1",
          );
       })->then( sub {
-         ( $event_id ) = @_;
+         ( $event_id_1 ) = @_;
```
Comment (leonerd) [inline tests/31sync/04timeline.pl]: The logic in these might read a little clearer, by way of associating the `$event_id_*` variables with their respective messages, by doing

       ...
       matrix_send_room_text_message( $user, $room_id,
          body => "Test message 1",
       )->on_done( sub { ( $event_id_1 ) = @_ } )
   })->then( sub {
       matrix_send_room_text_message( $user, $room_id,
          body => "Test message 2",
       )->on_done( sub { ( $event_id_2 ) = @_ } )
       ...

**Comment by leonerd** [comment]
LGTM as it stands. Or adjust it to use the `->on_done` suggestion. I'll leave that up to you.

**Comment by richvdh** [comment]
Thanks. I'll leave it for now, as none of the other tests in this area work in the way you suggest. It's a good idea though; I'll bear it in mind for the future.

**Pull request merged**